### PR TITLE
Feat : add button component

### PR DESCRIPTION
**Pull request by ZuberDunge**
btn icons
floting btn
buttons outlined
btn icons

### CODE REVIEW

**Comment by siddhigate** [comment]
Good to go!

**Comment by dubeyaditya29** [comment]
LGTM

**Review by junaidshaikh-js — COMMENTED** [review]
Good to go 🚀

File: docs/Components/button.css [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+.bat-btn-icon:hover {
+    background-color: var(--btn-icon-hover);
+}
+
+/* with icon */
+.bat-btn-with-icon {
+    display: flex;
+    gap: 0.75rem;
```
Comment (dikshadhote) [inline docs/Components/button.css:195]: Recommended to use px instead of rem for values less than 12 or 16 px

File: docs/Components/button.css [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    color: var(--prussian-blue-only);
+}
+
+.bat-btn-dark-outlined:hover {
+    background-color: var(--primary-red-dark);
+    color: var(--white-only);
+}
+
+.bat-btn-icon {
+    height: 3rem;
```
Comment (Hamdrive) [inline docs/Components/button.css:172]: Suggestion to use CSS variables with pre-defined `rem` values, can easily be used across all components!

File: docs/Components/button.css [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    background-color: var(--powder-blue);
+    padding: 0.75rem;
+    display: inline-block;
+    transition: .3s all;
+    font-weight: 500;
+    font-size: 1.25rem;
+}
+
+.bat-btn-disabled {
+    cursor: no-drop;
```
Comment (Hamdrive) [inline docs/Components/button.css:184]: why not use `cursor: not-allowed` instead?

File: docs/Components/button.css [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    gap: 2rem;
+    display: flex;
+    flex-direction: column;
+}
+
+.bat-btn-list {
+    gap: 1rem;
+    display: flex;
+    flex-wrap: wrap;
+}
```
Comment (Hamdrive) [inline docs/Components/button.css:220]: Add extra new line
Refer: [https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log](url)

**Review by Hamdrive — COMMENTED** [review]
Very clean and minimal code!
Suggestions given are optional but can be beneficial for overall CL

**Review by m26unkwn — APPROVED** [review]
Good to go

File: docs/Components/button.css (hunk 2 above)
Comment (ZuberDunge) [reply docs/Components/button.css:172]: Okay, I'll try from next time.

File: docs/Components/button.css (hunk 3 above)
Comment (ZuberDunge) [reply docs/Components/button.css:184]: actually, vs code auto-completed this and this work same as not allowed. idk which one to use semantically, I'll look into it

File: docs/Components/button.css (hunk 4 above)
Comment (ZuberDunge) [reply docs/Components/button.css:220]: Damn, i added line, but prettify removed the extra line.

**Pull request merged**